src/poetry/installation/executor.py

```diff
@@ -667,7 +668,7 @@ def _download_link(self, operation: Install | Update, link: Link) -> Path:
     @staticmethod
     def _validate_archive_hash(archive: Path, package: Package) -> str:
         file_dep = FileDependency(package.name, archive)
-        archive_hash: str = "sha256:" + file_dep.hash()
+        archive_hash: str = "sha256:" + get_file_hash(file_dep.full_path)
```

Comment:
Can we just pass `archive` to `get_file_hash()` and skip creating a `FileDependency`?

Reply:
**Short answer:** 
AFAICT that would be fine. Locally I made the change and it passes the tests and works when I tried installing something.

**Longer answer:**
Looking into it, the first thing that jumps out is that `FileDependency` does some checking around the `path` to make sure it is valid and also guarantees that `full_path` is absolute. 

Looking at [_download_link](https://github.com/python-poetry/poetry/blob/8c2113ce9b3c279337830fbce97ec413e8e152f4/src/poetry/installation/executor.py#L643), which is the only place `_validate_archive_hash` is called, `archive` seems like it will always be absolute, and will always be a valid file. So I think the checks in `FileDependency` are probably redundant...but it might be good if someone else takes a look to make sure I'm not missing something.

Reply:
I've looked this over again and do believe there are no problems doing this, so I've made the changes.